Project Management Process.md

```diff
@@ -0,0 +1,168 @@
+- Status
+  - What is the current status of this goal. Is it to-be-reviewed? In progress? Blocked? Done?
+- Dependencies (where applicable)
+- Priority (optional, priority can be defined in the backlog)
+- Any additional information that helps to reach this goal (optional)
+
+TODO: a clear example
+
+### Backlog
+Backlog is the collection of all goals in a project. It should be as comprehensive as possible, including everything that needs to happen or should happen in a particular project. The backlog indicates priority by order: the items that have the highest priority are at the top of the list. The backlog is owned and maintained by the project lead. As part of daily work, the backlog gets updated and a general grooming should happen on weekly basis.
+
+A backlog has the following information:
+- All goals of a project
+- Priority of goals in sorted order
+
+While the backlog is usually one big list of goals, it may sometimes become convoluted. In this case, the backlog list can be split in two: Backlog and Icebox. The icebox section lists goals that will probably not be worked on anytime soon, or are not ready to be worked on yet, whereas the backlog itself contains the goals that are ready or "approved". If split, together they are called "the backlog" and icebox is referenced only in its specific meaning. The split and maintenance of both lists are done the project lead.
+
+### Milestones
+A milestone is a group of project's goals. Each project has a set of milestones. A milestone is completed when all goals associated with it have been completed. Milestones are tracked in the project roadmap by the project lead.
+
+A milestone has the following information:
+- Name
+  - TODO: describe what is a good name, give an example
+- Due date
+- List of goals attached to it
+- Progress indicator
+
+The Project Lead keeps current milestones up to date on weekly-basis. New milestones should be generated and old milestones updated before the quarterly planning meeting. Prepare new milestones in good time before quarterly planning.
+
+TODO: a clear example
+
+### Project Roadmap
+The milestones are tracked in project's roadmap. The project lead owns the roadmap and is responsible for generating, updating and tracking the milestones. To make sure the roadmap is up to date, the project lead should go through the current milestones on weekly basis.
+
+The Roadmap Document contains the following information:
+- List of current milestones
+  - Ordered by expected due date, next upcoming milestone at the top
+- Current status of the project, ie. milestone progress
+- List of previous, recent milestones (< 6 months)
+- List of links to very old milestones (> 6 months)
+
+The roadmap can be a high-level overview of milestones and project's progress, so details suchs as list of goals per milestone can be omitted. However, if omitted, the roadmap must contain links to the detailed break down.
```

Comment:
Agreed! A tool to automate all of this would be amazing. Something where you specify which repos are "a project" and it'll generate roadmaps (md files perhaps?) and all based on the issues in the repos. Also, updates and sync milestones, labels, documents and whatnot across the project's repos. Or perhaps it has a dashboard that generates all that information in real-time!